aten/src/ATen/native/TensorShape.cpp

```diff
@@ -115,10 +115,14 @@ Tensor expand_as(const Tensor& self, const Tensor& other) {
 Tensor narrow(const Tensor& self, int64_t dim, int64_t start, int64_t length) {
   AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
   auto cur_size = self.size(dim);
-  if (start < 0 || start >= cur_size) {
+  if (start < 0) {
     AT_ERROR("start out of range");
   }
+#ifndef USE_TH_SIZE_ZERO_DIM
```

Comment:
I can't remember if `defined(USE_TH_SIZE_ZERO_DIM)` means that we are doing the new, good behavior, or not. Do we have a comment saying which way it is? (I will now proceed to read the code until it's clear which one is which.)

Reply:
Clearly `defined(USE_TH_SIZE_ZERO_DIM)` being true means the new behavior, based on the diff. I guess I was confused because it says `TH` and to me TH is synonymous with legacy.

Reply:
Yes, the name is bad, in particular because I thought I would only have to change TH/THC, but as you can see from here, there is ATen code as well.  Maybe rename to `USE_ZERO_SIZE_DIMS`?

Reply:
Or, maybe, `ALLOW_ZERO_SIZE_DIMS`.